components/community-groups/CommunityGroups.jsx

```diff
@@ -0,0 +1,171 @@
+import * as React from 'react';
+import { withStyles, makeStyles } from '@material-ui/core/styles';
+import { CircularProgress, Grid } from '@material-ui/core';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+
+import { db } from '../../src/firebase/firebaseSetup.js';
+import CommunityGroup from './CommunityGroupsCard';
+import CommunityGroupModel from '../../src/models/CommunityGroup';
+const styles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1,
+  },
+  progress: {
+    margin: theme.spacing(2),
+  },
+  indicator: {
+    display: 'none',
+  },
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: 120,
+  }
+}));
+
+class CommunityGroups extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      current: {},
+      showEvent: false,
+      tab: "Freshmen Male",
+      eventID: '',
+      loading: true,
+
+      chick: []
+    };
+
+    db.collection('communitygroups').get().then(((querySnapshot) => {
+        // TODO: add movement filter || section by movement
+      let current = {};
+      let freshmanGuy = [];

```

Comment:
I'm concerned about going down this path for filtering. Since not all campuses or movements organize their CGs the same way. Some do it just by grade, others just by gender. I would prefer something more dynamic for filtering even if it is a bit slower.

Reply:
Agree, Filters are dynamic and can be optional.